Automattic/wp-calypso
# Reader: move list streams to Redux

The current user's Reader lists are currently stored in Flux stores, and also partly in Redux state since #4036.

This PR moves list streams over to Redux completely, including following state for public lists.

There are no functional changes, so list streams should function exactly as they currently do on production - it's just the underpinnings that are changing.

Fixes #4166.

### To test

Visit a list stream (e.g. http://calypso.localhost:3000/read/list/blowery/comics) and ensure that:
- the list details (title and description) are correctly displayed
- for public lists not owned by you - following and unfollowing the list works as expected
- any changes in followed and unfollowed lists are reflected in the sidebar

<img width="785" alt="screen shot 2016-03-30 at 23 15 26" src="https://cloud.githubusercontent.com/assets/17325/14141832/5b2c1238-f6cd-11e5-8608-8f1902b9ea18.png">
